### PR TITLE
Refactor ToString instances

### LINKED ISSUE #1584: Refactor ToString Implementations
Inside `ToString.flix` we have e.g.:

```
instance ToString[Char] {
    def toString(x: Char): String = Char.toString(x)
}
```

but this means that `ToString` depends on `Char`-- but I would rather have it the other way. 

Thus move the impl. from `Char` into `ToString`:

```
instance ToString[Char] {
    def toString(x: Char): String = 
        import java.lang.Character:toString(Char);
        c.toString() as & Pure
}
```

and then change `Char` to:

```
namespace Char {
  ...
  def toString(c: Char): String = ToString.toString(c)
}
```

This applies to all instances in `ToString`.

Does it make sense?

### PR DESCRIPTION
**Pull request by esbenbjerre**
Fixes #1584.

### CODE REVIEW

**Comment by esbenbjerre** [comment]
Should be ready for merge.

File: main/src/library/BigInt.flix [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 namespace BigInt {
 
+    use ToString.toString;
```
Comment (magnus-madsen) [inline main/src/library/BigInt.flix:19]: Actually, can you just have this use locally? or perhaps even better, just write `ToString.toString`?

Comment (esbenbjerre) [reply main/src/library/BigInt.flix:19]: For `BigInt` specifically or all the files?

Comment (magnus-madsen) [reply main/src/library/BigInt.flix:19]: All :)

**Pull request merged**